accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,327 @@
+
+NOTE: So far, those systems that are registered as "Foreign/ANSIBLE" are not necessary been ever contacted by Uyuni yet, this means we do not have the real `machine-id`, which is not part of the Ansible inventory, and which needed to do a proper matching while onboarding the new minion. This means, before triggering the "Bootstrap" of an Ansible client, the `machine-id` needs to be properly set to the registered system. Easily done by(executing a command on the Ansible system before executing the "boostrap" state.
+
+For those systems that are "Foreign/ANSIBLE" we should enable some "Migration to Minion" tab that allows the user to trigger the bootstrap states via "salt-ssh" to convert this Ansible system into a fully-featured minion.
+
+
+## Maintain your Ansible infrastructure using Uyuni
+This section is more like the next level of the Ansible integration in Uyuni. So far, we have been focus on visualize your Ansible infrastrucutre in Uyuni and so some basic operations, like triggering playbooks in the control node or migrate to minion.
+
+This sections exposes lot of different possibilities in case that we really want to make Uyuni an UI for mantaining your playbooks and Ansible infrastructure. Maybe this is not really want we want for Uyuni, since there might be already better tools for this and it's opening a whole new world. In any case, some ideas that might be explored are:
+
+- Maintain your own Ansible Playbooks catalog in the Uyuni server: Playbook catalog (like Configuration State Channels)
+  * This would require, of course UI and DB investment. The idea would be to maintain the Ansible catalog inside the Uyuni server and push the playbook directly to the control node or the Ansible systems.
+  * Some questions: Which inventory / hosts groups and Ansible control node to use here for running the playbooks? Uyuni server as the Ansible control node?
+
+- Ansible Playbooks with Forms
+  * Similarly to what we currently have for Salt with "Formulas with Forms". Prefilled playbooks + some metadata to render the forms to filling the required information. This rendered playbook can be exposed via Salt fileserver as described above so the corresponding control node is able to fetch it.
+
+
+#### New UI tabs & pages
+
+- TBD
+
+# Defining a MVP
+[mvp]: #mvp
+
+This is a suggestion of implementation roadmap:
+
+### Step 0: The basic: gathering from an Ansible control node
+- Implement "ansible-gatherer" to gather from external host (SSH key based).
+- UI to add Ansible control nodes (reused from "Virtual Host Manager").
+- Push Ansible package to SLE15 clients tools to allow having SLE control nodes.
+- Inventory can be synced with Uyuni (add "Foreign/ANSIBLE" and "Salt/ANSIBLE", ...)
+
+### Step 1: Operating an Ansible control node
+- Trigger the playbooks from Ansible control node using "salt-ssh".
+- Transition from "Foreign/ANSIBLE" to "Salt/ANSIBLE".
```

Comment:
When I look at this, I have the feeling that only these 2 steps together are the MVP. But we could also remove some topics if it would help to make the implementation easier (not sure about it)

- Ansible for SLE15 is more a business decission then an implementation requirement. At the first place we want to get "existing" Ansible landscapes integrated. And this means there is a control node somewhere.
- "FOREIGN" should be discussed. If it would make things easier I would only go with "Salt/Ansible".

Reply:
> When I look at this, I have the feeling that only these 2 steps together are the MVP. 

Agreed

> But we could also remove some topics if it would help to make the implementation easier (not sure about it)
> 
>     * Ansible for SLE15 is more a business decission then an implementation requirement. At the first place we want to get "existing" Ansible landscapes integrated. And this means there is a control node somewhere.

There are reasons to provide the package, we can discuss on Monday.

>     * "FOREIGN" should be discussed. If it would make things easier I would only go with "Salt/Ansible".

We agreed no foreign and not traditional, only Salt+Ansible (but remember that's for the Ansible controllers!)